drivers/pinctrl/pinctrl_xlnx_zynqmp.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "pinctrl_soc.h"
+#include <zephyr/dt-bindings/pinctrl/pinctrl-zynqmp.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/syscon.h>
+#include <zephyr/devicetree.h>
+
+LOG_MODULE_REGISTER(pinctrl_xlnx_zynqmp, CONFIG_PINCTRL_LOG_LEVEL);
+
+#define DT_DRV_COMPAT xlnx_pinctrl_zynqmp
```

Comment:
driver does not use any DT APIs, other than the redundant line below.

Reply:
Now it does use DT API to fetch the MIO Register base address. Should it be removed despite that?